Source/Scene/ImageryProvider.js

```diff
@@ -340,11 +340,11 @@ define([
 
         var resource = Resource.createIfNeeded(url);
 
-        if (ktxRegex.test(resource)) {
+        if (ktxRegex.test(resource.url)) {
             return loadKTX(resource);
-        } else if (crnRegex.test(resource)) {
+        } else if (crnRegex.test(resource.url)) {
             return loadCRN(resource);
-        } else if (defined(imageryProvider.tileDiscardPolicy)) {
+        } else if (defined(imageryProvider) && defined(imageryProvider.tileDiscardPolicy)) {
```

Comment:
Why was the `imageryProvider.tileDiscardPolicy` check added?

Reply:
I didn't add `defined(imageryProvider.tileDiscardPolicy)` I added `defined(imageryProvider)` so people can provide null imageryProviders. 

Although, I am not sure this is the correct approach we should be using. We could either:
* Set ready to true by default and then pass a reference of imageryProvider to `loadImage`. 
* Move the loadImage occur in the requestImage function and set ready to true by default.

Reply:
Sorry, not enough caffeine yet.  In the place you added the call for this, you have the ImageryProvider, so you can just pass it into this function instead of changing this function.

Reply:
Talked to @ProjectBarks offline.  After looking into it a little more, I think this is the cleanest option for now (and it would have just crashed previously so it's not like we were guaranteeing any sort of behavior).  Long term, we should probably unify ktx/crtn/resource image loading a bit more than it is now.

Reply:
Wrote up #8070, not a priority right now.